app-modules/notification/src/Notifications/Channels/EmailChannel.php

```diff
@@ -123,14 +124,35 @@ public static function afterSending(object $notifiable, OutboundDeliverable $del
         }
     }
 
-    public function canSendWithinQuotaLimits(Notification $notification, object $notifiable): bool
+    public static function determineQuotaUsage(array $recipients): int
+    {
+        return collect($recipients)->filter(function ($recipient) {
+            $user = User::where('email', $recipient->getAddress())->first();
```

Comment:
Should we add a `with('roles')` here to eager load the roles relationship for the Users?